File: challenge_15.rb
def find_factorial(n)
    return 1 if n == 0
    return n * find_factorial(n-1)
 end

def lattice_paths(rows,cols)
  find_factorial(rows+cols)/(find_factorial(rows)*find_factorial(cols))
end

p lattice_paths(20,20)

File: challenge_15/challenge_15.exs
defmodule Lattice do
	def paths(rows,cols), do: round(fact(rows+cols)/(fact(rows)*fact(cols)))
	def fact(n), do: Enum.reduce((1..n), 1, fn x,y -> x * y end)
end

IO.inspect(Lattice.paths(20,20))